### DISCUSSION TITLE
Qualité de vie de développeur

### DISCUSSION THREAD

**Discussion by MelvinCou** [Crownicles/Crownicles#2468] [Nouvelles idées]
Salut,

Je suis le projet depuis un bon moment et j'ai des propositions qui peuvent améliorer la QoL des devs sur le projet :

- Mettre en place un [devcontainer](https://containers.dev/)
  - Container (docker) de développement => contient tout les outils nécessaire de dev sans avoir à les installer
  - Intégration avec les [PR github](https://docs.github.com/fr/codespaces/developing-in-a-codespace/using-github-codespaces-for-pull-requests) => permet de faire des revues de code sans avoir à mettre en pause le dev d'une autre fonctionnalité en cours
- Utiliser [Bun](https://bun.sh/) en temps qu'environnement d'exécution au lieu de Node
  - Performances : bun est écrit en zig, un langage compilé, alors que node est écrit en js
  - Prise en charge native du ts
  - Recharge à la volée (hot reload) : actuellement, j'ai l'impression que le programme est relancé manuellement avec `start` du package.json. C'est assez frustrant pour un langage interprété
  - [Bundling](https://bun.sh/docs/bundler) : compilation du projet dans un fichier js compact, ça permet d'éviter d'avoir un énorme node_module de 500+ Mo en prod
  - ⚠️ Impact sur le code : même si bun se veut être un remplacement de node, [certaines fonctionnalités propre à node ne sont pas implémentés](https://bun.sh/docs/runtime/nodejs-apis). De plus, le bundling peut avoir un impact sur le code suivant la manière dont il est structuré
  - ⚠️ Prise en charge partielle de windows, il vaut mieux utiliser la WSL ou un devcontainer pour le dev

Ce sont des choses que je peux aider à mettre en place. Est-ce que cela vous intéresse ?

**Comment by BastLast** [comment]
Plop, merci beaucoup pour le petit com !

Réponse courte car jai pas énorme de temps,
Dev container, why not mais perso pour les avoir utilisés j'ai jamais ete très convaincu, ca fait gagner du temps sur des trucs mais je préfère que l'équipe reste tous sur le meme setup principal (aka webstorm), et le setup est suffisamment simple pour que ca soit pas un soucis

**Reply by MelvinCou** [reply]
Pour ce qui est de l'intégration des devcontainers avec les ide jetbrains, c'est vrai qu'elle n'est pas tip top comparé à celle de vscode.

**Comment by BastLast** [comment]
Pour bun, j'avais vu passer aussi et j'avoue ne pas avoir testé, la non feature parity avec node me fais peur mais faudrait tester pour être sûr.
De manière générale j'ai un peu une giga flemme de chercher à optimiser 5sec persu pdt le dev, ma prio actuelle c'est de finir le rewrite de la v5.

Tldr, pas un non définitif mais ces 2 suggestions sont pas du tout dans le top de ma todolist

**Comment by BastLast** [comment]
Pour ce qui est des review, on manque de temps de dev deja pour etre très consciencieux donc je pense pas que ca nous aide beaucoup d'avoir un dev container.

**Comment by BastLast** [comment]
Pour bun, why not faire une experiment mais si ya plus de 10 min de taf je bloque perso

**Reply by MelvinCou** [reply]
Je vais faire un poc à partir de la branche master pour voir ce que ça donne.

**Reply by BastLast** [reply]
> Je vais faire un poc à partir de la branche master pour voir ce que ça donne.

Nah pas master, DraftBot v5, master est obsolète

**Reply by MelvinCou** [reply]
Re

Je suis en train de setup draftbot. Il me manque juste le `websocket_url` pour la config de discord. Qu'est-ce que je dois mettre ?

**Reply by BastLast** [reply]
http://localhost:7071 devrait marcher :) draftbot v5 est encore en dev, tout n'est pas encore final en term de doc etc

btw si tu veux nous aider on a besoin de dev, ça serait encore plus top de bosser ensemble sur les sujets prios de la v5 plutot que sur bun (après je veux pas t'empecher de faire ce qui t'amuse, juste si tu veux nous aider au mieux, on a plus besoin d'aide sur autre chose :) )

**Reply by MelvinCou** [reply]
Je ne suis pas très doué pour tout ce qui touche les technos JS, mais je peux toujours essayer.

Maintenant j'ai une erreur avec les migrations SQL :

```js
{ event: 'migrating', name: '002-elo.js' }
MigrationError: Migration 002-elo.js (up) failed: Original error: (conn=5, no: 1054, SQLState: 42S22) Unknown column 'effect' in 'field list'
sql: SELECT `id`, `keycloakId`, `health`, `fightPointsLost`, `score`, `weeklyScore`, `level`, `experience`, `money`, `class`, `badges`, `guildId`, `topggVoteAt`, `nextEvent`, `petId`, `lastPetFree`, `effect`, `effectEndDate`, `effectDuration`, `mapLinkId`, `startTravelDate`, `notifications`, `gloryPoints`, `gloryPointsLastSeason`, `fightCountdown`, `rage`, `updatedAt`, `createdAt` FROM `players` AS `Player`; - parameters:[]
```

Le champ effect est bien répertorié dans la liste, pb de typage ?

**Reply by BastLast** [reply]
alors là j'ai pas le temps de t'aider immédiatement, je check ça ce soir

**Reply by MelvinCou** [reply]
J'ai créé des PR avec du refacto mineur sur mon fork  https://github.com/MelvinCou/DraftBot/pulls.
Si elles t'intéressent, je les créées ici.

**Reply by BastLast** [reply]
> J'ai créé des PR avec du refacto mineur sur mon fork  https://github.com/MelvinCou/DraftBot/pulls.
> Si elles t'intéressent, je les créées ici.

Jai check, ya des trucs qui me plaisent, go les envoyer sur le main (sauf le dev container perso jai pas envie de maintenir ça)

Si on peut switch la conv sur le discord du bot ca serait cool aussi, les com d'une issue c'est pas idéal

**Reply by MelvinCou** [reply]
Ok pas de soucis.

Je suis déjà sur le discord Officiel.
hadeath. (Oui avec le point)
